app/code/Magento/CustomerGraphQl/Model/Customer/ValidateCustomerData.php

```diff
@@ -0,0 +1,61 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+declare(strict_types=1);
+
+namespace Magento\CustomerGraphQl\Model\Customer;
+
+use Magento\Framework\GraphQl\Exception\GraphQlInputException;
+
+/**
+ * Class ValidateCustomerData
+ */
+class ValidateCustomerData
+{
+    /**
+     * Get allowed/required customer attributes
+     *
+     * @var GetAllowedCustomerAttributes
+     */
+    private $getAllowedCustomerAttributes;
+
+    /**
+     * ValidateCustomerData constructor.
+     *
+     * @param GetAllowedCustomerAttributes $getAllowedCustomerAttributes
+     */
+    public function __construct(GetAllowedCustomerAttributes $getAllowedCustomerAttributes)
+    {
+        $this->getAllowedCustomerAttributes = $getAllowedCustomerAttributes;
+    }
+
+    /**
+     * Validate customer data
+     *
+     * @param array $customerData
+     * @param $addRequiredAttributes
+     *
+     * @return void
+     *
+     * @throws GraphQlInputException
+     */
+    public function execute(array $customerData, $addRequiredAttributes = false): void
+    {
+        $attributes = $this->getAllowedCustomerAttributes->execute(array_keys($customerData), $addRequiredAttributes);
+        $errorInput = [];
+
+        foreach ($attributes as $attributeInfo) {
+            if ($attributeInfo->getIsRequired() && empty($customerData[$attributeInfo->getAttributeCode()])) {
```

Comment:
Looks like we will have failed validation if the value is `0` (possible variant in some cases)

Reply:
Fixed